src/server/pps/cmds/cmds_test.go

```diff
@@ -830,3 +834,85 @@ func TestPipelineBuildMissingPath(t *testing.T) {
 // 	os.Args = []string{"pachctl", "create", "pipeline", "--push-images", "-f", "test-push-images.json"}
 // 	require.NoError(t, rootCmd().Execute())
 // }
+
+func runPipelineWithImageGetStderr(t *testing.T, image string) string {
+	// reset and create some test input
+	require.NoError(t, tu.BashCmd(`
+		yes | pachctl delete all
+		pachctl create repo in
+		pachctl put file -r in@master:/ -f ../../../../etc/testing/pipeline-build/input
+	`).Run())
+
+	cmd := exec.Command("/bin/bash", "-c", fmt.Sprintf(`
```

Comment:
My opinion on this is mixed, but I think the way to do this that's most consistent with the other command tests would be to use the `match` binary; something like:
```
tu.BashCmd(`
pachctl create pipeline <<EOF 2>&1 >/dev/null \
  | match "expected error"
...pipeline text...
EOF
`)
```

I think for consistency's sake it might be worth changing, but let me know what you think.

Reply:
Hmm.

I think I prefer the current approach for two reasons:

1. It asserts that the warning comes out of stderr, rather than "stdout or stderr". Warnings should come out of stderr ([ref](https://stackoverflow.com/questions/1430956/should-i-output-warnings-to-stderr-or-stdout#:~:text=For%20those%20reasons%2C%20in%20general%2C%20you%20output%20warnings%20to%20stderr.&text=The%20warning%20should%20go%20to,is%20an%20issue%20of%20flexibility.)), and it's nice that the tests assert that.
2. One of the tests wants to assert a negative - that there is no warning. That's (a) tricky to do with `match` and (b) would require pushing the logic up into the `exec.Command` command body which is currently factored out (and unfactoring this would make the tests overall have more duplication in them).

I'd be happy to move the "run this command and capture stderr" code into `testutil` though, as that would be more idiomatic than calling `exec.Command` directly from the test suite. What do you think?

I pushed a separate commit which makes `create pipeline` error fail the test by just checking for the error.

I don't feel super strongly about it, and if you'd prefer to have more consistency between tests by using the `match` tool I'm happy to update the tests to do that. Let me know 😄

Reply:
I think it's fine to leave this as-is, your argument makes sense.

> I'd be happy to move the "run this command and capture stderr" code into testutil though, as that would be more idiomatic than calling exec.Command directly from the test suite. What do you think?

I think this is probably the right long-term answer, though I also don't think it makes sense to block this PR on it. I filed an issue for it here: https://github.com/pachyderm/pachyderm/issues/5174, in case we're motivated to work on it later.